From ef2f198a69dcd707c8101ab1e31b8360ac27c952 Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Mon, 18 Nov 2013 12:49:48 -0500
Subject: [PATCH 55/65] Ticket 47597 - Convert retro changelog plug-in to betxn

Retro cl plugin is already betxn aware.  The template and 20betxn.pl script
needed to be updated to reflect the new default.

https://fedorahosted.org/389/ticket/47597

Reviewed by: richm(Thanks!)
(cherry picked from commit 3dca85ec629be641f07ae2ecfef59609d4dc88e2)
(cherry picked from commit 229d270428dc4cfabd7d367444f1c0b10a60ef87)
---
 ldap/admin/src/scripts/20betxn.pl | 4 +++-
 ldap/ldif/template-dse.ldif.in    | 1 +
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/ldap/admin/src/scripts/20betxn.pl b/ldap/admin/src/scripts/20betxn.pl
index 2c56707..6f9b5e1 100644
--- a/ldap/admin/src/scripts/20betxn.pl
+++ b/ldap/admin/src/scripts/20betxn.pl
@@ -12,10 +12,12 @@ sub runinst {
     #     cn=Multimaster Replication Plugin
     #     cn=Roles Plugin,cn=plugins,cn=config
     #     cn=USN,cn=plugins,cn=config
+    #     cn=Retro Changelog Plugin,cn=plugins,cn=config
     my @objplugins = (
         "cn=Multimaster Replication Plugin,cn=plugins,cn=config",
         "cn=Roles Plugin,cn=plugins,cn=config",
-        "cn=USN,cn=plugins,cn=config"
+        "cn=USN,cn=plugins,cn=config",
+        "cn=Retro Changelog Plugin,cn=plugins,cn=config"
     );
     foreach my $plugin (@objplugins) {
         my $ent = $conn->search($plugin, "base", "(cn=*)");
diff --git a/ldap/ldif/template-dse.ldif.in b/ldap/ldif/template-dse.ldif.in
index 95ed60c..12df7b6 100644
--- a/ldap/ldif/template-dse.ldif.in
+++ b/ldap/ldif/template-dse.ldif.in
@@ -557,6 +557,7 @@ cn: Retro Changelog Plugin
 nsslapd-pluginpath: libretrocl-plugin
 nsslapd-plugininitfunc: retrocl_plugin_init
 nsslapd-plugintype: object
+nsslapd-pluginbetxn: on
 nsslapd-pluginenabled: off
 nsslapd-pluginprecedence: 25
 nsslapd-plugin-depends-on-type: database
-- 
1.8.1.4

